models/uniswap/base/uniswap_v3_base_trades.sql

```diff
@@ -0,0 +1,109 @@
+{{ config(tags=['dunesql'],
+    schema = 'uniswap_v3_base',
+    alias = alias('trades'),
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
+    post_hook='{{ expose_spells(\'["base"]\',
+                                "project",
+                                "uniswap_v3",
+                                \'["wuligy"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2023-07-16' %}
+
+WITH dexs AS
+(
+    --Uniswap v3
+    SELECT
+        t.evt_block_time AS block_time
+        ,t.recipient AS taker
+        ,CAST(NULL as VARBINARY) as maker
+        ,CASE WHEN amount0 < INT256 '0' THEN abs(amount0) ELSE abs(amount1) END AS token_bought_amount_raw -- when amount0 is negative it means trader_a is buying token0 from the pool
+        ,CASE WHEN amount0 < INT256 '0' THEN abs(amount1) ELSE abs(amount0) END AS token_sold_amount_raw
+        ,NULL AS amount_usd
+        ,CASE WHEN amount0 < INT256 '0' THEN f.token0 ELSE f.token1 END AS token_bought_address
+        ,CASE WHEN amount0 < INT256 '0' THEN f.token1 ELSE f.token0 END AS token_sold_address
+        ,t.contract_address as project_contract_address
+        ,t.evt_tx_hash AS tx_hash
+        ,t.evt_index
+    FROM
+    {{ source('uniswap_v3_base', 'UniswapV3Pool_evt_Swap') }} t
+    INNER JOIN 
+    {{ source('uniswap_v3_base', 'UniswapV3Factory_evt_PoolCreated') }} f
+        ON f.pool = t.contract_address
+    {% if is_incremental() %}
+    WHERE t.evt_block_time >= date_trunc('day', now() - interval '7' day)
+    {% endif %}
+)
+SELECT
+    'base' AS blockchain
+    ,'uniswap' AS project
+    ,'3' AS version
+    ,TRY_CAST(date_trunc('DAY', dexs.block_time) AS date) AS block_date
+    ,CAST(date_trunc('month', dexs.block_time) AS date) AS block_month
+    ,dexs.block_time
+    ,erc20a.symbol AS token_bought_symbol
+    ,erc20b.symbol AS token_sold_symbol
+    ,case
+        when lower(erc20a.symbol) > lower(erc20b.symbol) then concat(erc20b.symbol, '-', erc20a.symbol)
+        else concat(erc20a.symbol, '-', erc20b.symbol)
+    end as token_pair
+    ,dexs.token_bought_amount_raw / power(10, erc20a.decimals) AS token_bought_amount
+    ,dexs.token_sold_amount_raw / power(10, erc20b.decimals) AS token_sold_amount
+    ,CAST(dexs.token_bought_amount_raw AS UINT256) AS token_bought_amount_raw
+    ,CAST(dexs.token_sold_amount_raw AS UINT256) AS token_sold_amount_raw
+    ,coalesce(
+        dexs.amount_usd
+        ,(dexs.token_bought_amount_raw / power(10, p_bought.decimals)) * p_bought.price
+        ,(dexs.token_sold_amount_raw / power(10, p_sold.decimals)) * p_sold.price
```

Comment:
similar questions here

Reply:
The reason for the null values in 
 - token_bought_amount
 - token_sold_amount
 ... 
is that the token information is missing in the token.erc20 table. 
As a result, the decimals value cannot be obtained, leading to the values being set as null.

Perhaps it would be advisable to upload a set of Erc20 Token information for the Base chain. This way, the issue should be resolved.

Reply:
yes, you are correct. the base erc20 token spell has started as of a few days ago:
https://github.com/duneanalytics/spellbook/blob/main/models/tokens/base/tokens_base_erc20.sql

you are welcome to add more in this PR, as you see fit

Reply:
i am also okay with merging without the extra tokens, as the code still works, and you can add to tokens file over time. when tokens is edited, these models and dex trades will be refreshed to reflect those new values

Reply:
Yes, I submitted a pr today to supplement the ERC20 tokens information with the Base chain. 
It includes 756 tokens from the Base chain. 

Here is the link: https://github.com/duneanalytics/spellbook/pull/4135

Reply:
ah, nice, thank you! will look to merge at same time